plugins/bulk-import/src/hooks/useRepositories.ts

```diff
@@ -182,8 +191,8 @@ export const useRepositories = (options: {
   const debouncedUpdateResources = useDebounceCallback(prepareData, 200);
 
   React.useEffect(() => {
-    debouncedUpdateResources?.(value, loading);
-  }, [debouncedUpdateResources, value, loading]);
+    debouncedUpdateResources?.(value, isQueryLoading);
+  }, [debouncedUpdateResources, value, isQueryLoading]);
 
   return useDeepCompareMemoize({
```

Comment:
I see an issue that the pagination forward and backend shows always a loading indicator.

I guess that the `useRepository` doesn't use the caching benefits of react-query yet/correctly.

My recommendation is that you remove almost all useStates from this hook.

If you need calculate results, based on the useQuery hook, you can use `useMemo` instead.

The code should also get rid off the effect to call debouncedUpdateResources and the deepCompareMemoize function.

The order should be:

1. useQuery to fetch data
2. useMemo if you must transform "complex" data with a dependency just on the data.